test/framework/landscaper/commonframework.go

```diff
@@ -0,0 +1,379 @@
+	flag.StringVar(&newCfg.ComponentConfig.baseUrl, "componentBaseUrl", "", "the base url of the component descriptor (e.g. eu.gcr.io/gardener-project/landscaper/dev/components)")
+	flag.StringVar(&newCfg.ComponentConfig.name, "componentName", "", "the name of the component descriptor (e.g. github.com/gardener/gardener)")
+	flag.StringVar(&newCfg.ComponentConfig.version, "componentVersion", "", "the version of the component descriptor (e.g. 1.19.0)")
+
+	landscaperCommonConfigFromFlags = newCfg
+
+	return landscaperCommonConfigFromFlags
+}
+
+func validateLandscaperCommonConfig(cfg *LandscaperCommonConfig) {
+	if cfg == nil {
+		ginkgo.Fail("no landscaper common framework configuration provided")
+	}
+
+	if !StringSet(cfg.KubeconfigPathLandscaperCluster) {
+		ginkgo.Fail("Need to set the path to the kubeconfig of the cluster where the landscaper is installed (flag: -landscaper-kubecfg-path)")
+	}
+
+	if !StringSet(cfg.ComponentConfig.name) {
+		ginkgo.Fail("Need to set the name of the component descriptor (e.g. github.com/gardener/gardener) (flag: componentName) ")
+	}
+
+	if !StringSet(cfg.ComponentConfig.baseUrl) {
+		ginkgo.Fail("Need to set the base url of the component descriptor (e.g. eu.gcr.io/gardener-project/landscaper/dev/components) (flag: componentBaseUrl) ")
+	}
+
+	if !StringSet(cfg.ComponentConfig.version) {
+		ginkgo.Fail("Need to set the version of the component descriptor (e.g. 1.19.0) (flag: componentVersion) ")
+	}
+}
+
+func mergeLandscaperCommonConfig(base, overwrite *LandscaperCommonConfig) *LandscaperCommonConfig {
+	if base == nil {
+		return overwrite
+	}
+
+	if overwrite == nil {
+		return base
+	}
+
+	if StringSet(overwrite.KubeconfigPathLandscaperCluster) {
+		base.KubeconfigPathLandscaperCluster = overwrite.KubeconfigPathLandscaperCluster
+	}
+
+	if StringSet(overwrite.ComponentConfig.name) {
+		base.ComponentConfig.name = overwrite.ComponentConfig.name
+	}
+
+	if StringSet(overwrite.ComponentConfig.version) {
+		base.ComponentConfig.version = overwrite.ComponentConfig.version
+	}
+
+	if StringSet(overwrite.ComponentConfig.baseUrl) {
+		base.ComponentConfig.baseUrl = overwrite.ComponentConfig.baseUrl
+	}
+	return base
+}
+
+func (f *CommonFrameworkLandscaper) GetInstallation(name string) *landscaperv1alpha1.Installation {
+	name = fmt.Sprintf("%s-%s", name, f.ResourceSuffix)
+
+	return &landscaperv1alpha1.Installation{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: f.Config.TargetNamespace,
+		},
+		Spec: landscaperv1alpha1.InstallationSpec{
+			Blueprint: landscaperv1alpha1.BlueprintDefinition{
+				Reference: &landscaperv1alpha1.RemoteBlueprintReference{
+					// the component descriptor has to contain the blueprint as a resource
+					// with type:blueprint and name: blueprint
+					// in the future, this could be made configurable
+					ResourceName: "blueprint",
```

Comment:
think we should name the blueprint `gardenlet-blueprint` as there could be multiple blueprints in teh future

Reply:
makes sense.